homeassistant/helpers/entity.py

```diff
@@ -380,8 +380,9 @@ def _async_write_ha_state(self):
                 state = str(round(temp) if prec == 0 else round(temp, prec))
                 attr[ATTR_UNIT_OF_MEASUREMENT] = units.temperature_unit
         except ValueError:
-            # Could not convert state to float
-            pass
+            # Could not convert state to float - avoid passing along invalid
+            # unit_of_measurement in this case
+            attr[ATTR_UNIT_OF_MEASUREMENT] = None
```

Comment:
Removing the unit of measurement during the runtime will most likely break or have a least side effects on the integrations that exports the data to a third party system like InfluxDB or Prometheus.

Reply:
How would it break when the measurement is not even a number? It makes no sense for me to log a unit when there's no value measured in that unit. An alternative might be to convert the unit for temperature measurements even if the value is not a number. This seems wrong to me (again), but if it's less likely to break external things, I guess I could live with it. Any other ideas?